inspire_json_merger/postprocess.py

```diff
@@ -199,8 +199,12 @@ def _additem(item, object, path):
                     new_path, object[idx] = _additem(item, object[idx], path[1:])
             except ValueError:
                 if len(path) == 1:
-                    object[path[0]] = thaw(item)
-                    return (path[0],), object
+                    if isinstance(item, list) or not isinstance(object[path[0]], list):
+                        object[path[0]] = thaw(item)
+                        return (path[0],), object
+                    elif item not in object[path[0]]:
+                        object[path[0]].append(thaw(item))
+                        return (path[0], (len(object[path[0]]) - 1)), object
```

Comment:
what's going on here?

Reply:
This is a fix to a bug I noticed when I was investigating this task. Problem here is that if we have a `path` which is just straight path without any index number (for example "authors") we overwrite whole path in merged data with update which is really wrong. So for example we have: {"a":[1,2,3,4]} and we want add an item [5] to "a" but we won't provide a proper path with index where we want to store new value we will overwrite whole [1,2,3,4] with [5]. Now with the fix we check if this is a list or not, and if it is then we add new element at the end instead of overwriting whole list.

Reply:
could you maybe add a few unit tests for this function? it's become huge.

Reply:
:+1:

Reply:
Ok, now this function have 100% coverage in tests from tests_postprocess.py